test/middleware/express.test.ts

```diff
@@ -0,0 +1,341 @@
+                        schema,
+                        { url: '' }
+                    )
+                ).not.toThrow();
+            });
+
+            xtest('...Fixed Window', () => {
+                expect(
+                    expressRateLimitMiddleware(
+                        'FIXED_WINDOW',
+                        { refillRate: 1, bucketSize: 10 }, // FIXME: Replace with valid params
+                        schema,
+                        { url: '' }
+                    )
+                ).not.toThrow();
+            });
+
+            xtest('...Sliding Window', () => {
+                expect(
+                    expressRateLimitMiddleware(
+                        'SLIDING_WINDOW_LOG',
+                        { refillRate: 1, bucketSize: 10 }, // FIXME: Replace with valid params
+                        schema,
+                        { url: '' }
+                    )
+                ).not.toThrow();
+            });
+
+            xtest('...Sliding Window Counter', () => {
+                expect(
+                    expressRateLimitMiddleware(
+                        'SLIDING_WINDOW_COUNTER',
+                        { refillRate: 1, bucketSize: 10 }, // FIXME: Replace with valid params
+                        schema,
+                        { url: '' }
+                    )
+                ).not.toThrow();
+            });
+        });
+
+        test('Throw an error for invalid schemas', () => {
+            const invalidSchema: GraphQLSchema = buildSchema(`{Query {name}`);
+
+            expect(
+                expressRateLimitMiddleware('TOKEN_BUCKET', {}, invalidSchema, { url: '' })
+            ).toThrowError('ValidationError');
+        });
+    });
+
+    describe('Middleware is Functional', () => {
+        // Before each test configure a new middleware amd mock req, res objects.
+        beforeAll(() => {
+            jest.useFakeTimers('modern');
+        });
+
+        afterAll(() => {
+            jest.useRealTimers();
+        });
+
+        beforeEach(() => {
+            middleware = expressRateLimitMiddleware(
+                'TOKEN_BUCKET',
+                { refillRate: 1, bucketSize: 10 },
+                schema,
+                {}
+            );
+            mockRequest = {
+                query: {
+                    // complexity should be 2 (1 Query + 1 Scalar)
+                    query: `Query {
+                    scalars: {
+                        num
+                    }
+                `,
+                },
+                ip: '123.456',
+            };
+
+            mockResponse = {
+                json: jest.fn(),
+                send: jest.fn(),
+                sendStatus: jest.fn(),
+                locals: {},
+            };
+
+            complexRequest = {
+                // complexity should be 10 if 'first' is accounted for.
+                // scalars: 1, droid: 1, reviews (4 * (1 Review, 1 episode))
```

Comment:
Enums are equivalent to scalarn and have a weight of zero, not 1, so this won't work as expected.